tests/tooltip.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { test, expect } from '@playwright/test';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/tooltip-test.html');
+});
+
+test('Confirms tooltip text is visible upon hover', async ({ page }) => {
+  const tooltip = page.getByRole('tooltip', { name: 'This is the hover test'})
+  await expect(tooltip).not.toBeVisible()
+  const wordToHover = page.getByText('Hover Test').first()
+  await wordToHover.hover();
+  await expect(tooltip).toBeVisible()
+});
+
+test('Confirms tooltip text is above text to hover if location attribute is specified as "top"', async ({ page }) => {
+  const textLocatedBelow = await page.locator('p:near(:text("This is the top test"))').textContent()
+  expect(textLocatedBelow).toBe('Top Test')
+});
+
+test('Confirms tooltip text is below text to hover if location attribute is specified as "bottom"', async ({ page }) => {
```

Comment:
Additionally, the ticket calls for tests that confirm the position of the tooltip is to the top, bottom, left, and right of the hover text depending on the specified location attribute in the v-tooltip.  However, location based selectors weren't working for me.  Do you know of another way to test for location besides using location based selectors like (await page.locator('p:above(:text("This is the right test"))').textContent()).

Reply:
not optimal, but are there particular classes on inner elements that determine their relative positions?

Reply:
I haven't tried this yet with playwright but my first intuition for a test like this would be to compare the viewport offset of the tooltip against its target with an api like `getBoundingClientRect()`. Not sure if it's available in this environment though.

Reply:
That totally worked!!! Thank you both!